test/functional/tests/stress/test_stress_change_io_class_config_io.py

```diff
@@ -0,0 +1,178 @@
+#
+# Copyright(c) 2022 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+import os
+import random
+import threading
+import pytest
+
+from datetime import timedelta
+from time import sleep
+from api.cas import casadm
+from api.cas.cache_config import CacheMode
+from api.cas.ioclass_config import IoClass
+from core.test_run_utils import TestRun
+from storage_devices.disk import DiskTypeSet, DiskType, DiskTypeLowerThan
+from test_tools.disk_utils import Filesystem
+from test_tools.fio.fio import Fio
+from test_tools.fio.fio_param import IoEngine
+from test_utils.asynchronous import start_async_func
+from test_utils.size import Size, Unit
+from tests.io_class.io_class_common import generate_and_load_random_io_class_config
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_stress_io_class_change_config_during_io_raw():
+    """
+    title: Set up IO class configuration from file during IO - stress.
+    description: |
+        Check Open CAS ability to change IO class configuration during running IO
+        on small cache and core devices.
+    pass_criteria:
+        - No system crash
+        - IO class configuration changes successfully
+        - No IO errors
+    """
+    cores_per_cache = 4
+
+    with TestRun.step("Prepare devices."):
+        cache_device = TestRun.disks['cache']
+        core_device = TestRun.disks['core']
+
+        cache_device.create_partitions([Size(150, Unit.MebiByte)])
+        core_device.create_partitions([Size(256, Unit.MebiByte)] * cores_per_cache)
+
+        cache_device = cache_device.partitions[0]
+
+    with TestRun.step("Start cache in Write-Back mode and add core devices."):
+        cache = casadm.start_cache(cache_device, cache_mode=CacheMode.WB, force=True)
+        cores = [cache.add_core(part) for part in core_device.partitions]
+
+    with TestRun.step("Create IO class configuration file for 33 IO classes with random allocation "
+                      "and priority value."):
+        generate_and_load_random_io_class_config(cache)
+
+    with TestRun.step("Run IO for all CAS devices."):
+        fio_task = start_async_func(run_io, cores, True)
+
+    with TestRun.step("In two-second time interval change IO class configuration "
+                      "(using random values in allowed range) and cache mode "
+                      "(between all supported). Check if Open CAS configuration has changed."):
+        change_mode_thread = threading.Thread(target=change_cache_mode, args=[cache, fio_task])
+        change_io_class_thread = threading.Thread(target=change_io_class_config,
+                                                  args=[cache, fio_task])
+        change_mode_thread.start()
+        sleep(1)
+        change_io_class_thread.start()
+
+        while change_io_class_thread.is_alive() and change_mode_thread.is_alive():
```

Comment:
Shouldn't it be `or` instead of `and`? If one of the threats was still alive and the test started cleanup it would lead to unwanted fail